src/core/lib/security/credentials/s2a/grpc_s2a_credentials_options.h

```diff
@@ -58,7 +64,7 @@ struct grpc_s2a_credentials_options {
                     const std::vector<char*>& target_service_account_list);
 
  private:
-  char* handshaker_service_url_;
+  const char* handshaker_service_url_;
```

Comment:
It looks like this class takes ownership of `handshaker_service_url_`. Is there a better way to make this clear in the code, e.g., using unique pointers?

Reply:
Also since we can use C++, maybe we can use something easier to handle than `char*`.

Reply:
I've changed all of the `char*`'s in `grpc_s2a_credentials_options.{h,cc}` to `std::string`.